lib/wildcard-matcher.js

```diff
@@ -0,0 +1,62 @@
+const escapeStringRegexp = require('escape-string-regexp')
```

Comment:
Should we `'use strict'` at the top of all our JS files?

nit: A relatively brief top-comment at the top of each file giving its general intent would be helpful. E.g. could mention this implements a common "WildcardMatcher" spec from apm.git (though that repo doesn't have a good link for that spec, just the JSON file of tests).

Reply:
_nod_ -- seems reasonable -- will add.